### ISSUE TITLE
Cannot get New Tab Tools working in FF 56.0b3 (64-bit) Win 7

### ISSUE THREAD

**Issue by Hellishly** [darktrojan/newtabtools#377]
It was working just fine until I did a reboot about 2 hours ago. 
Now when I open the NTT tab it is blank.
I have tried uninstalling NTT, closing FF, re-installing NTT. I've also tried to start over and set the tab up again from scratch, but it won't save any URL's I add, nor will it show any boxes. It just stays blank.
This is what I see: https://gyazo.com/02751c2bfe3849e8564e5cfbd2b0cad6
https://gyazo.com/5074be2c0a2dcf2bf1afb642ae38a737

At this point I'm at a total loss on what to do.

Oh and I'm pretty much tech not savvy. (In other words I need stuff explained as simply and thoroughly as possible).

**Comment by DanaMW** [comment]
I don't mean to butt in on your repo BUT i run 6 browsers on two machines with your extension. I had the same problem with one browser and I resolved it by importing a predefined one. What I am suggesting is creating a newtabtools.zip with just a few defined. and make it available to see if that works for him. As it did for me. Me out.

**Comment by artworksmetal** [comment]
Hellish, I had the exact same issue. Except I didn't even get the NTT box to come up. I think uninstalling then reinstalling can really mess with Firefox sometimes.
Hopefully you have made a backup which you should be able to restore.
If you don't have a backup, or the backup doesn't work, my next suggestion would be to do a refresh of Firefox.
https://support.mozilla.org/en-US/kb/refresh-firefox-reset-add-ons-and-settings
Good luck and keep us posted

**Comment by Hellishly** [comment]
https://github.com/artworksmetal  "I think uninstalling then reinstalling can really mess with Firefox sometimes."
I'm thinking you misunderstood. I only uninstalled and reinstalled after trying many other suggested ideas to try to get back my NTT. In other words Firefox had already messed up (deleted, lost, whatever it did) my NTT.

I have no idea what you mean about a backup. A backup of what?
I really don't want to have to do a refresh of FF as that removes everything, which means trying to find stuff that is Legacy in some cases. (I did read about, and almost did do a Refresh, till I realised getting everything back may not be so easy).

**Comment by Hellishly** [comment]
https://github.com/DanaMW  " I had the same problem with one browser and I resolved it by importing a predefined one. What I am suggesting is creating a newtabtools.zip with just a few defined. and make it available to see if that works for him. As it did for me."

I have NO idea what you are talking about. Could you please elaborate?

**Comment by DanaMW** [comment]
The extension can export and import the configuration. I cleared up my issue by importing my backed up configuration. The backup is default named newtabtools.zip so i suggested that the author make available a backup to allow you to try importing it.

**Comment by artworksmetal** [comment]
I'm just guessing here but if you are still using Legacy Firefox that may be the source of your problem.

Another possibility is that your Homepage is not set correctly. Try this:

1. Open Firefox
2. Select Tools/Options from the menu.
3. Change When Firefox starts two "Show a Blank Page."
4. Close then reopen Firefox.
5. Click on a new tab (+ sign) to open NTT . If New Tab Tools looks like it opens properly then you're in luck. Continue the steps below.   If it doesn't open properly then I have run out of suggestions.
6. Close the blank new tab page.
7. Go to the right hand NTT page and select Tools/Options from the menu.
8. Select "Show your home pages" and click on the "Use Current Pages" button. The long list of numbers and letters will change when you click it.
Now your homepage should properly point to the NTT extension.

**Comment by Hellishly** [comment]
https://github.com/artworksmetal  Not sure what you meant by Legacy Firefox. The version I was using is in the subject line.
I followed your instructions, still no NTT.

Finally I updated FF just to see as when I re-installed NTT it was the January 8th version. This brought the boxes back, but I'm having to reconfigure it all again from scratch.

So to sum up. updating Firefox brought the add-on back, without any of the boxes I'd saved.
Why it broke is anyone's guess as all I did was reboot, same as I'd done every other 4th, or 5th day. Nothing had changed, no updates, nothing. It just broke.

**Comment by artworksmetal** [comment]
I'm sorry you had to go through all that.
Firefox 56 is the last release to support older, or 'legacy' extensions. @darktrojan may have to chime in here, because I don't know exactly how NTT  might be different if you have not upgraded to FF 57.
NTT (at least the newest version) has a backup feature that allows you to save the boxes into a zip file. You should do that often while you are reconfiguring. Save it somewhere convenient like your desktop.

**Comment by darktrojan** [comment]
Hi @Hellishly. I don't know what's happened here. From your first comment it looks like Firefox has lost or broken your New Tab Tools settings. NTT should be able to start up anyway and give you *something* usable, what you have is a weird state indeed.

I don't think you're going to be able to recover what you had before, unfortunately. :-(

(As a side note, I suggest you switch to the Firefox "release" channel, unless you have a specific reason to be on the "beta" channel, especially if you're not updating regularly/automatically. [Downloading again](https://www.mozilla.org/en-US/firefox/) and installing will put you back on the release channel.)

**Comment by Hellishly** [comment]
https://github.com/darktrojan
I got it going again after updating to Firefox 58.6. 
I ended up on the Beta channel about 2-3 years back when I had serious stability issues with a release version. The Beta fixed it. Normally I'm up to date, but I'd slipped behind recently, and as everything was working perfectly I left it go.

I couldn't recover anything from my previous configuration of NTT. But have it all set up again, and it's running smoothly.

Thanks all for the assistance. I'll close this now as it's resolved.
